java/arcs/tools/GenerationUtils.kt

```diff
@@ -0,0 +1,109 @@
+package arcs.tools
+
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.buildCodeBlock
+
+/**
+ * Utility to translate [List]s into a code-generated [List] collection.
+ *
+ * Will generate an empty collection when appropriate.
+ *
+ * @param template KotlinPoet template string to apply to each item in the collection.
+ * @return generated [CodeBlock] of a [List].
+ */
+fun List<*>.toGeneration(template: String = "%L") =
+    toGeneration { builder, item -> builder.add(template, item) }
+
+/**
+ * Utility to translate [List]s into a code-generated [List] collection.
+ *
+ * Will generate an empty collection when appropriate.
+ *
+ * @param template callback that combines a [CodeBlock.Builder] with an item in the collection.
+ * @return generated [CodeBlock] of a [List].
+ */
+fun <T> List<T>.toGeneration(
+    template: (builder: CodeBlock.Builder, item: T) -> Unit
+) = buildCodeBlock {
+    if (this@toGeneration.isEmpty()) {
+        add("emptyList()")
+        return build()
```

Comment:
Is it okay to return build() here, but not at the end of the buildCodeBlock block?  I think the better approach is to just use `return@buildCodeBlock`